source/slang/slang-ir-link.cpp

```diff
@@ -1519,6 +1520,146 @@ static void diagnoseUnresolvedSymbols(TargetRequest* req, DiagnosticSink* sink,
     }
 }
 
+void convertAtomicToStorageBuffer(
+    IRSpecContext* context, 
+    Dictionary<int, List<IRInst*>>& bindingToInstMapUnsorted)
+{
+    IRBuilder builder = *context->builder;
+
+    for (auto& bindingToInstList : bindingToInstMapUnsorted)
+    {
+        //I don't need a sorted list, just the max so I can make an array of uint[(maxOffset/4)+1]
+        int64_t maxOffset = 0;
+        for (auto& i : bindingToInstList.second)
+        {
+            int64_t currOffset = int64_t(i->findDecoration<IRGLSLOffsetDecoration>()->getOffset()->getValue());
+            maxOffset = (maxOffset < currOffset) ? currOffset : maxOffset;
+        }
+        auto instToSwitch = *bindingToInstList.second.begin();
+        builder.setInsertBefore(instToSwitch);
+        
+        // now we know the max offset and can construct a new storageBuffer with key of type maxOffset/4 (32bit uint per atomic_uint)
+        auto elementType = builder.getArrayType(
+            builder.getUIntType(),
+            builder.getIntValue(builder.getUIntType(), (maxOffset / sizeof(uint32_t))+1)
+        );
+
+        // now we make our struct which goes within with some name (does not matter), 
+        StringBuilder nameStruct;
+        nameStruct << "atomic_uints";
+        nameStruct << bindingToInstList.first;
+        nameStruct << "_t";
+        nameStruct << "_paramGroup";
+        auto structType = builder.createStructType();
+        builder.addNameHintDecoration(structType, nameStruct.produceString().getUnownedSlice());
+
+        // our "_data" struct member
+        auto elementBufferKey = builder.createStructKey();
+        builder.addNameHintDecoration(elementBufferKey, UnownedStringSlice("_data"));
+        auto elementBufferType = elementType;
+        auto _dataField = builder.createStructField(structType, elementBufferKey, elementBufferType);
+        
+        // make our valid storage buffer type
+        auto std430 = builder._createInst(sizeof(IRTypeLayoutRules), builder.getType(kIROp_Std430BufferLayoutType), kIROp_Std430BufferLayoutType);
+        IRGLSLShaderStorageBufferType* storageBuffer;
+        {
+            IRInst* ops[] = { structType, std430 };
+            storageBuffer = builder.createGLSLShaderStorableBufferType(2, ops);
+        }
+
+        // link storage buffer to any known atomic_uint
+        instToSwitch->setFullType(storageBuffer);
+        
+        // now we need to make all call's to:
+        // 1. convert all atomic_uint function type parameters to a inlined call of atomic_uint->inout int ref of 
+        // field->element-addr based on location(offset) from the atomic_uint->storage_buffer logic above
```

Comment:
I see "1." but I don't see "2.".
And this needs to be either removed or simplified to be helpful.

Reply:
Please remove this comment.

Reply:
I will simplify the comment as per the first review, 

these comments are here since the code here was trying to follow how slang-ir-append-consume-struct was commented since it similarly is pulling a similar style of 'magic trick' to manipulate var A into var B and var C  (it has some 'whats' rather than 'whys')